axelrod/tests/unit/test_titfortat.py

```diff
@@ -272,3 +272,50 @@ def test_rounds(self):
         self.versus_test(axelrod.OmegaTFT(), axelrod.Alternator(),
                          [C, C, D, C, D, C, C, C, D, C, C, C, D, D, D, D, D, D],
                          [C, D, C, D, C, D, C, D, C, D, C, D, C, D, C, D, C, D])
+
+
+class TestGradual(TestPlayer):
+
+    name = 'Gradual'
+    player = axelrod.Gradual
+    expected_classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_strategy(self):
+        """Starts by cooperating."""
+        self.first_play_test(C)
+
+    def test_effect_of_strategy(self):
+        """Punishes defection with a growing number of defections and calms the opponent
+            with two Cooperations in a row"""
+        self.responses_test([C], [C], [C])
```

Comment:
I believe `responses_test` can take an `attr` dictionary that will check for attribute values. That is, you can pass `attr = {"calming": True}` as an argument and at the end of the sequence of plays the test function will check that these attributes have been set on the player. Just an FYI for test writing in the future. Look at the tests for `Grudger` for some examples.